scripts/upload-source-maps.js

```diff
@@ -1,65 +1,88 @@
  */
-
 const version = process.argv[2]
-const packages = [
-  { name: 'logs', service: 'browser-logs-sdk' },
-  { name: 'rum', service: 'browser-rum-sdk' },
-  { name: 'rum-slim', service: 'browser-rum-sdk' },
-]
+let uploadPathTypes = process.argv[3].split(',')
+const siteByDatacenter = {
+  us1: 'datadoghq.com',
+  eu1: 'datadoghq.eu',
+  us3: 'us3.datadoghq.com',
+  us5: 'us5.datadoghq.com',
+  ap1: 'ap1.datadoghq.com',
+}
 const sitesByVersion = {
   staging: ['datad0g.com', 'datadoghq.com'],
   canary: ['datadoghq.com'],
-  v4: ['datadoghq.com', 'datadoghq.eu', 'us3.datadoghq.com', 'us5.datadoghq.com', 'ap1.datadoghq.com'],
+  // TODO remove in next major
+  v4: Object.values(siteByDatacenter),
 }
 
 runMain(() => {
-  for (const { name, service } of packages) {
-    const bundleFolder = `packages/${name}/bundle`
-    renameFilesWithVersionSuffix(bundleFolder, name)
-    for (const site of sitesByVersion[version]) {
-      const normalizedSite = site.replaceAll('.', '-')
-      const apiKey = getSecretKey(`ci.browser-sdk.source-maps.${normalizedSite}.ci_api_key`)
+  if (uploadPathTypes.find((value) => value === 'root')) {
+    // root upload done last to rename source files safely
+    uploadPathTypes = uploadPathTypes.filter((value) => value !== 'root')
+    uploadPathTypes.push('root')
+  }
 
-      uploadSourceMaps(site, apiKey, name, service, bundleFolder)
+  for (const { packageName, service } of packages) {
+    const bundleFolder = path.dirname(buildSourcePath(packageName))
+    for (const uploadPathType of uploadPathTypes) {
+      let sites
+      let buildUploadPath
+      if (uploadPathType === 'root') {
+        sites = sitesByVersion[version]
+        buildUploadPath = buildRootUploadPath
+        renameFilesWithVersionSuffix(packageName, bundleFolder)
+      } else {
+        sites = [siteByDatacenter[uploadPathType]]
+        buildUploadPath = buildDatacenterUploadPath(uploadPathType)
+      }
+      const prefix = path.dirname(`/${buildUploadPath(packageName, version)}`)
+      uploadSourceMaps(packageName, service, prefix, bundleFolder, sites)
     }
   }
   printLog('Source maps upload done.')
 })
 
-function renameFilesWithVersionSuffix(bundleFolder, packageName) {
+function renameFilesWithVersionSuffix(packageName, bundleFolder) {
   // The datadog-ci CLI is taking a directory as an argument. It will scan every source map files in
   // it and upload those along with the minified bundle. The file names must match the one from the
   // CDN, thus we need to rename the bundles with the right suffix.
-  for (const ext of ['js', 'js.map']) {
-    const sourceFilePath = `${bundleFolder}/datadog-${packageName}.${ext}`
-    const targetFilePath = `${bundleFolder}/datadog-${packageName}-${version}.${ext}`
+  for (const extension of ['js', 'js.map']) {
+    const sourceFilePath = buildSourcePath(packageName, extension)
+    const targetFilePath = `${bundleFolder}/${buildRootUploadPath(packageName, version, extension)}`
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏  I find this way harder to understand than before 😅 Granted, we are using nice function to avoid repeating ourselves, but still, it's hard to understand that we rename the file in the same folder with just a suffix added.

Reply:
wdyt of ed2991a725fc85e886d96ff061b8347c233d965a?

Reply:
That's better!